pkg/auxnetdevice/auxNetDeviceProvider_test.go

```diff
@@ -0,0 +1,192 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2022 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package auxnetdevice_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/stretchr/testify/mock"
+
+	"github.com/jaypipes/ghw"
+	"github.com/jaypipes/pcidb"
+
+	"github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pkg/auxnetdevice"
+	"github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pkg/factory"
+	"github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pkg/types"
+	tmocks "github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pkg/types/mocks"
+	"github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pkg/utils"
+	"github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pkg/utils/mocks"
+)
+
+var _ = Describe("AuxNetDeviceProvider", func() {
+	Describe("getting new instance of auxNetDeviceProvider", func() {
+		Context("with correct arguments", func() {
+			It("should return valid instance of the provider", func() {
+				rf := &tmocks.ResourceFactory{}
+				p := auxnetdevice.NewAuxNetDeviceProvider(rf)
+				Expect(p).ToNot(BeNil())
+			})
+		})
+	})
+	Describe("getting devices", func() {
+		Context("when there are none", func() {
+			rf := &tmocks.ResourceFactory{}
+			p := auxnetdevice.NewAuxNetDeviceProvider(rf)
+			config := &types.ResourceConfig{}
+			dDevs := p.GetDiscoveredDevices()
+			devs := p.GetDevices(config)
+			It("should return empty slice", func() {
+				Expect(dDevs).To(BeEmpty())
+				Expect(devs).To(BeEmpty())
+			})
+		})
+	})
+	Describe("adding 3 target devices", func() {
+		Context("when 2 are valid devices, but only one have auxiliary devices", func() {
+			fs := &utils.FakeFilesystem{
+				Dirs: []string{
+					"sys/bus/auxiliary/devices",
+					"sys/bus/pci/devices/0000:00:00.1/foo.bar.0",
```

Comment:
can we have examples here (throughout this file) based on real data where applicable ?

Reply:
Ok